### ISSUE TITLE
Massive `quote-props` false positive in v5.3.1

### ISSUE THREAD

**Issue by zloirock** [eslint-stylistic/eslint-stylistic#963]
<img width="766" height="791" alt="Image" src="https://github.com/user-attachments/assets/8b51555a-8ee1-4125-b226-a89ea186ea69" />

for example, on

```js
$({ target: 'WeakMap', stat: true, forced: true }, {
  from: createCollectionFrom(WeakMapHelpers.WeakMap, WeakMapHelpers.set, true) // <- !!!
});
```
Config: https://github.com/zloirock/core-js/blob/master/tests/eslint/eslint.config.js

**Comment by 9romise** [comment]
Thank you for raising this issue, and for your ongoing maintenance of `core-js`.

This problem occurred because the list of keywords was updated in this version, while the `keywords` option in `quote-props` was enforcing quotes around keywords. This is because, prior to ES6, the keywords were not allowed as property names.  

You can temporarily set `keywords` to `false` to avoid this error.  

If you think it’s necessary, we could also adjust the rule to only check pre-ES6 keywords.  

I look forward to your thoughts and discussion on this.

**Comment by zloirock** [comment]
> This is because, prior to ES6, the keywords were not allowed as property names.

Those words are allowed keys in all ES versions.

**Comment by zloirock** [comment]
ES3:
<img width="710" height="715" alt="Image" src="https://github.com/user-attachments/assets/0e10f619-8b71-4d5b-861d-93db3736a8f9" />

**Comment by 9romise** [comment]
> > This is because, prior to ES6, the keywords were not allowed as property names.
> 
> Those words are allowed keys in all ES versions.

Sorry for the unclear wording, the keywords I referred to here are not the newly added keywords, but specifically the reserved words.

I know what you mean now. I'll revert this rule to only check ES3 keywords as discussed.
